### PR TITLE
feat: [sc-11043] [Referrals][frontend] - ENS name stretch goal

### PR DESCRIPTION
**Pull request by marcinciarka**
Story details: https://app.shortcut.com/oazo-apps/story/11043

### CODE REVIEW

File: blockchain/ens.ts [hunk 1]
```diff
@@ -17,3 +20,69 @@ export function resolveENSName$(context$: Observable<Context>, address: string)
     }),
   )
 }
+
+// address => ens name
+export async function addressToEnsNameMainnet(ensName: string) {
+  const provider = new ethers.providers.JsonRpcBatchProvider(
+    getNetworkRpcEndpoint(NetworkIds.MAINNET),
+  )
+  return await provider.lookupAddress(ensName).catch((err: Error) => {
```
Comment (piotrwitek) [inline blockchain/ens.ts:29]: I would suggest to clean those awaits from the return statements ,async functions always returns promises So those awaits are redundant

File: blockchain/ens.ts [hunk 2]
```diff
@@ -17,3 +20,69 @@ export function resolveENSName$(context$: Observable<Context>, address: string)
     }),
   )
 }
+
+// address => ens name
+export async function addressToEnsNameMainnet(ensName: string) {
```
Comment (piotrwitek) [inline blockchain/ens.ts:25]: I'm really confused what this function does, it accept ens name as argument but it's supposed to map address to the ens name also The comment above suggest it accept address and return ens name, that means the argument name is Incorrect

File: blockchain/ens.ts [hunk 3]
```diff
@@ -17,3 +20,69 @@ export function resolveENSName$(context$: Observable<Context>, address: string)
...
+    getNetworkRpcEndpoint(NetworkIds.MAINNET),
+  )
+  return await provider.lookupAddress(ensName).catch((err: Error) => {
+    console.warn(`Error looking up ENS name for address: ${ensName}/${err.message}`)
+    return null
+  })
+}
+
+// ens name => address
+export async function ensNameToAddressMainnet(address: string) {
```
Comment (piotrwitek) [inline blockchain/ens.ts:36]: here is the same issue as above regarding the argument name

File: blockchain/ens.ts [hunk 4]
```diff
@@ -17,3 +20,69 @@ export function resolveENSName$(context$: Observable<Context>, address: string)
...
+  const provider = new ethers.providers.JsonRpcBatchProvider(
+    getNetworkRpcEndpoint(NetworkIds.MAINNET),
+  )
+  return await provider.resolveName(address).catch((err: Error) => {
```
Comment (piotrwitek) [inline blockchain/ens.ts:40]: remove await

File: blockchain/ens.ts [hunk 5]
```diff
@@ -17,3 +20,69 @@ export function resolveENSName$(context$: Observable<Context>, address: string)
...
+      .catch((err: Error) => {
+        console.warn(`Error looking up ENS name for address: ${err.message}`)
+      })
+  }, [address])
+
+  return [ensName]
+}
+
+export const useMainnetEnsNames = (addresses?: string[]) => {
+  const firstAddressesList: { [key: string]: string } = {}
```
Comment (piotrwitek) [inline blockchain/ens.ts:64]: initialAddressesList

File: features/referralOverview/FeesView.tsx [hunk 6]
```diff
@@ -22,7 +24,8 @@ export interface UpsertUser {
 
 export function FeesView({ userReferral }: Props) {
   const { t } = useTranslation()
-
+  const [ensReferrerName] = useMainnetEnsName(userReferral?.referrer)
+  const ensReferredNames = useMainnetEnsNames(userReferral.referrals)
```
Comment (piotrwitek) [inline features/referralOverview/FeesView.tsx:28]: These api return values are not particularly consistent, first one is wrapped in an array and the other one is not. I would Suggest to strive for consistent Behavior For reusable modules as it is much easier to use those

**Review by piotrwitek — CHANGES_REQUESTED** [review]

File: blockchain/ens.ts (hunk 1 above)
Comment (marcinciarka) [reply blockchain/ens.ts]: Good catch.

File: blockchain/ens.ts (hunk 2 above)
Comment (marcinciarka) [reply blockchain/ens.ts]: 👌

File: blockchain/ens.ts (hunk 3 above)
Comment (marcinciarka) [reply blockchain/ens.ts]: 👌

File: blockchain/ens.ts (hunk 4 above)
Comment (marcinciarka) [reply blockchain/ens.ts]: 👌

File: features/referralOverview/FeesView.tsx (hunk 6 above)
Comment (marcinciarka) [reply features/referralOverview/FeesView.tsx]: fixed

**Review by piotrwitek — APPROVED** [review]

**Pull request merged**